includes/classes/order.php

```diff
@@ -272,8 +272,8 @@ function cart() {
                                    from (" . TABLE_CUSTOMERS . " c, " . TABLE_ADDRESS_BOOK . " ab )
                                    left join " . TABLE_ZONES . " z on (ab.entry_zone_id = z.zone_id)
                                    left join " . TABLE_COUNTRIES . " co on (ab.entry_country_id = co.countries_id)
-                                   where c.customers_id = '" . (int)$_SESSION['customer_id'] . "'
-                                   and ab.customers_id = '" . (int)$_SESSION['customer_id'] . "'
+                                   where c.customers_id = '" . (int)(!empty($_SESSION['customer_id']) ? $_SESSION['customer_id'] : 0) . "'
+                                   and ab.customers_id = '" . (int)(!empty($_SESSION['customer_id']) ? $_SESSION['customer_id'] : 0) . "'
```

Comment:
For these cases I think the `(int)` should be moved inside, like:

```diff
- (int)(!empty($_SESSION['customer_id']) ? $_SESSION['customer_id'] : 0) 
+ (!empty($_SESSION['customer_id']) ? (int)$_SESSION['customer_id'] : 0) 
```

Reply:
Wasn't sure how best to "support" the upgrade process and also had a "concern" about someone choosing to use some non-integer style variable in lieu of the 0.  By placing the `(int)` up front, seemed sounder for such individual modification; however, also see that the 0 sufficiently meets the expectation.

I don't mind moving the `(int)` to inside, leaving any modifications that may be made to be properly implemented by the modifier/user.